### PR TITLE
Add XML parser

### PR DESCRIPTION
**Pull request by scomans**
Adds XML reader function to file system utility used by updated Jetbrains Toolbox extension (#1200).

### CODE REVIEW

**Comment by oliverschwendener** [comment]
I know that we have "readJsonFile" and "writeJsonFile" in the file system utility, but as we're introducing a new npm package here I would love to have the XML parsing in a separate module, similar to the `IniFileParser`. So if you have to read an xml file and parse it you would have to use the FileSystemUtility first to read the text file and then parse the XML using the XmlFileParser. Can you move the XML parsing out of the FileSystemUtility into it's separate module? Let me know if you need help setting up the module.

**Review by oliverschwendener — CHANGES_REQUESTED** [review]
See my comment above

**Comment by scomans** [comment]
Will do that tomorrow morning

**Comment by oliverschwendener** [comment]
I added the basic structure how I think it would suit best. I would be happy if you could have a look and add some description for the XmlParser.

**Comment by scomans** [comment]
Looks good to me

File: src/main/Core/XmlParser/Contract/XmlParser.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * Offers a method to parse XML content.
+ */
+export interface XmlParser {
+    /**
+     * Parses the given XML string and returns the result as an object.
+     * @param xml - The XML string to be parsed.
+     * @param [options] - Optional settings for parsing.
+     * @param options.preserveOrder - Whether to preserve the order of elements.
+     * @param options.ignoreAttributes - Whether to ignore attributes in the XML.
+     * @returns {T} - The parsed result.
```
Comment (oliverschwendener) [inline src/main/Core/XmlParser/Contract/XmlParser.ts:11]: Very nice, thanks! 💪

**Review by oliverschwendener — APPROVED** [review]

**Comment by oliverschwendener** [comment]
Would you like to be the code owner of this module? Then all future changes can be reviewed by you

**Comment by scomans** [comment]
Sure :)

**Review by oliverschwendener — APPROVED** [review]

**Pull request merged**